backend/app/analysis/text_parser.py

```diff
@@ -0,0 +1,165 @@
+"""
+Custom command to populate text model with definition, examples, and images
+"""
+import string
+import nltk
+import Levenshtein as Lev
+
+
+# Modified Code from Bing library
+def get_sentences(text):
+    """
+    This function, when given a string of words, breaks it apart into individual
+    sentences through markers of sentence breaks and line breaks(\n)
+
+    :param text: str, any text
+    :return: list containing the fragments of the broken down text
+    """
+    # breaks up text by line breaks first
+    lines = [p for p in text.split('\n') if p]
+    sentences = []
+
+    # breaks up each line by sentence markers
+    for line in lines:
+        sentences.extend(nltk.tokenize.sent_tokenize(line))
+
+    return sentences
+
+
+def correct_sentence(given_sent, correct_sent):
+    """
+    Function takes the user's sentence and the correct sentence and compares them to find missing
+    words, the incorrect words, and a grade
+
+    :param given_sent: str, sentence the user types into the text box
+    :param correct_sent: str, the correct sentence that the instructor inputs
+    :return: dictionary with parameters holding the missing words, correct words,
+        word/correctness at each index
+    """
+
+    grade = {}
+
+    # tokenize given_sent and correct_sent and turn them both into lists
+    given_tok = nltk.tokenize.word_tokenize(given_sent.lower().translate(
+        str.maketrans('', '', string.punctuation)))
+    correct_tok = nltk.tokenize.word_tokenize(correct_sent.lower().translate(
+        str.maketrans('', '', string.punctuation)))
```

Comment:
As a takeaway for the future, it's always worth considering how to make your code as readable as possible for the next developer. These long chained method calls with nested chained method certainly take me a second to parse fully, which is probably the impulse behind leaving a comment here. In this case, we could pull out the `maketrans` dictionary to improve readability:
```
translate_dictionary = str.maketrans('', '', string.punctuation)
given_tok = nltk.tokenize.word_tokenize(given_sent.lower().translate(translate_dictionary))
correct_tok = nltk.tokenize.word_tokenize(correct_sent.lower().translate(translate_dictionary))
```

Or even increase the verbosity even further to improve readability:
```
translation_dictionary = str.maketrans('', '', string.punctuation)
given_sanitized_sentence = given_sent.lower().translate(translation_dictionary)
correct_sanitized_sentence = correct_sent.lower().translate(translation_dictionary)
given_tokenized_sentence = nltk.tokenize.word_tokenize(given_sanitized_sentence)
correct_tokenized_sentence = nltk.tokenize.word_tokenize(correct_sanitized_sentence)
```

The above pattern does two things:
1. Allows the next developer to skim more quickly and, in effect, ignore the right side of the expression. The expressively named variables tell me all I need to know without even needing to read the implementation (though of course I'll read the implementation if I'm interested in understanding exactly what's going on).
2. Removes the need for an explicit comment. This is a personal opinion, but I often find comments are ways of getting around writing expressive code.